bazel-java-deps.bzl

```diff
@@ -7,52 +7,59 @@
 
 load("@rules_jvm_external//:defs.bzl", "maven_install")
 load("@rules_jvm_external//:specs.bzl", "maven")
+load("@scala_version//:index.bzl", "scala_major_version", "scala_version")
+
+version_specific = {
+    "2.12": [
+        "ai.x:diff_2.12:2.0.1",
+        "org.gnieh:diffson-spray-json_2.12:3.1.1",
+    ],
+}
 
 def install_java_deps():
     maven_install(
-        artifacts = [
-            "ai.x:diff_2.12:2.0.1",
+        artifacts = version_specific.get(scala_major_version, []) + [
             "ch.qos.logback:logback-classic:1.2.3",
             "ch.qos.logback:logback-core:1.2.3",
             "com.auth0:java-jwt:3.10.3",
             "com.auth0:jwks-rsa:0.11.0",
-            "com.chuusai:shapeless_2.12:2.3.3",
+            "com.chuusai:shapeless_{}:2.3.3".format(scala_major_version),
```

Comment:
Would it make sense to gather all Scala dependencies in a single array? Probably nothing to gain in practical terms, just asking.

Reply:
I’ve done it for the individual `da_scala_library`s but not here since this doesn’t change too often. I could definitely be convinced to do it though if you prefer it though. This was just simpler and leads to a cleaner diff.

Reply:
"Cleaner diff" is definitely a good point. We can make this is a separate PR if we think it's useful. 🙇🏻